examples/gameroom/daemon/src/rest_api/routes/submit.rs

```diff
@@ -64,39 +77,263 @@ pub fn submit_scabbard_payload(
+        "".to_string()
+    } else if invalid_batches.len() == 1 {
+        if let BatchStatus::Invalid(invalid_transactions) = &invalid_batches[0].status {
+            if invalid_transactions.len() <= 1 {
+                "A transaction failed. Please try again. If it continues to fail contact your administrator for help.".to_string()
+            } else {
+                "Several transactions failed. Please try again. If it continues to fail contact your administrator for help.".to_string()
+            }
+        } else {
+            "".to_string()
+        }
+    } else {
+        "Several transactions failed. Please try again. If it continues to fail please contact your administrator.".to_string()
+    }
+}
+
+fn check_batch_status(
+    client: web::Data<Client>,
+    splinterd_url: &str,
+    link: &str,
+    start_time: Instant,
+    wait: u64,
+) -> Box<dyn Future<Item = Vec<BatchInfo>, Error = RestApiResponseError>> {
+    let splinterd_url = splinterd_url.to_owned();
+    let link = link.to_owned();
+    debug!("Checking batch status {}", link);
+    Box::new(
+        client
+            .get(format!("{}{}", splinterd_url, link))
+            .send()
+            .map_err(|err| {
+                RestApiResponseError::InternalError(format!("Failed to send request {}", err))
+            })
+            .and_then(move |mut resp| {
+                let body = match resp.body().wait() {
+                    Ok(b) => b,
+                    Err(err) => {
+                        return Either::B(future::err(RestApiResponseError::InternalError(
+                            format!("Failed to receive response body {}", err),
+                        )))
+                    }
+                };
+                match resp.status() {
+                    StatusCode::OK => {
+                        let batches_info: Vec<BatchInfo> = match serde_json::from_slice(&body) {
+                            Ok(b) => b,
+                            Err(err) => {
+                                return Either::B(future::err(RestApiResponseError::InternalError(
+                                    format!("Failed to parse response body {}", err),
+                                )))
+                            }
+                        };
+
+                        // If batch status is still pending and the wait time has not yet passed,
+                        // send request again to re-check the batch status
+                        if batches_info
+                            .iter()
+                            .any(|batch_info| match batch_info.status {
+                                BatchStatus::Pending => true,
+                                BatchStatus::Valid(_) => true,
+                                _ => false,
+                            })
+                            && Instant::now().duration_since(start_time) < Duration::from_secs(wait)
+                        {
+                            sleep(Duration::new(1, 0));
```

Comment:
What's this sleep for?

Reply:
I thought it would be better to wait a little before sending the request again. So, that gameroomd doesn't end up sending too many requests to the batch status endpoint in case the batches take long to be committed.

Reply:
Ah, I guess a comment and using `Duration::from_secs(1)` might improve clarity.

Reply:
addressed